src/components/ProductDetail/ProductDetail.js

```diff
@@ -1,39 +1,85 @@
+import { motion } from 'framer-motion'
+import PropTypes from 'prop-types'
 import React from 'react'
 import { useHistory, useParams } from 'react-router-dom'
 import styled from 'styled-components'
 import { ReactComponent as Arrow } from '../../images/icons/back-arrow.svg'
 import { ReactComponent as Gluten } from '../../images/icons/gluten.svg'
 import { ReactComponent as Soy } from '../../images/icons/soy.svg'
-import { ReactComponent as Heart } from '../../images/icons/heart.svg'
-import PropTypes from 'prop-types'
-
+import Rating from '../Rating/Rating'
 
 ProductDetail.propTypes = {
   product: PropTypes.array,
   onFavoriteClick: PropTypes.func,
   favorites: PropTypes.array,
 }
 
-export default function ProductDetail({ product, onFavoriteClick, favorites }) {
+export default function ProductDetail({
+  product,
+  onFavoriteClick,
+  favorites,
+  pageTransitions,
+}) {
   const { id } = useParams()
-  const productItem = product?.filter(product => JSON.stringify(product.id) === id)
+
+  const productItem = product?.filter(
+    (product) => JSON.stringify(product.id) === id
+  )
   const history = useHistory()
-  
+
+  const variants = {
+    clicked: { scale: [1, 1.5, 1], x: [0, 20, 0] },
+  }
+
   function handleClick() {
     history.goBack()
   }
 
-  return (
-      <>
-    {productItem?.map((productItem) => (
-    <DetailWrapper key={productItem.id}>
+  return productItem?.map((productItem) => (
+    <DetailWrapper
+      initial="out"
+      animate="in"
+      exit="out"
+      variants={pageTransitions}
+      key={productItem.id}
+    >
       <ColoredBackground>
         <ArrowStyled onClick={handleClick} />
         <ProductName>{productItem.title}</ProductName>
       </ColoredBackground>
       <ImageWrapper>
-        <ProductImage src={productItem.image} alt={productItem.title}/>
-        <HeartStyled favored={favorites?.find(favoriteItem => favoriteItem.id === productItem.id)} onClick={() => onFavoriteClick(productItem)}/>
+        <ProductImage src={productItem.image} alt={productItem.title} />
+        <HeartStyled
```

Comment:
I don't know a thing about motion svg's, but if there is a opportunity to import it from somewhere else, I would take it.